### ISSUE TITLE
Some questions about VQ-KD

### ISSUE THREAD

**Issue by Basums** [microsoft/unilm#878]
Hello, when I read and reproduce your work, there is a consistent question about VQ-KD. When training MIM, it can be regarded as an offline teacher or Tokenizer, but can't it perform Imagenet classification task alone? Vq-kd is a Tokenizer that learns the semantic information from DINO or CLIP. It should be good for him to do classification directly, right?
Therefore, there is a question: why not use the Vit obtained from VQ-KD to do the classification task?
I am looking forward to your reply

**Comment by pengzhiliang** [comment]
Dear:

Thanks for the suggestion~
I will evaluate and report it after finished.

Zhiliang.

**Comment by Basums** [comment]
Dear,
I am glad to receive your timely reply. Regarding VQ-KD, there is a hypothesis about the tokenizer obtained from knowledge distillation learning, that this teacher is able to do the classification task, but the effect is not particularly good (lacking some masks), and I hope you can tell us after your experiment to verify this conjecture.
This issue can hopefully be closed after you report this conclusion, please.

**Comment by Basums** [comment]
Dear, let me add some questions, I see that among the experiments VQ-KD is only pre-trained on imagenet1k, why not on 21k, or is 1k pre-training enough, if this is the reason, it is really not worth to take this teacher for the classification task unless its trained on a larger scale or multiple iterations.

**Comment by pengzhiliang** [comment]
> Dear, I am glad to receive your timely reply. Regarding VQ-KD, there is a hypothesis about the tokenizer obtained from knowledge distillation learning, that this teacher is able to do the classification task, but the effect is not particularly good (lacking some masks), and I hope you can tell us after your experiment to verify this conjecture. This issue can hopefully be closed after you report this conclusion, please.

Of course. I have conducted experiments about it. And I will report the results here tomorrow.



> Dear, let me add some questions, I see that among the experiments VQ-KD is only pre-trained on imagenet1k, why not on 21k, or is 1k pre-training enough, if this is the reason, it is really not worth to take this teacher for the classification task unless its trained on a larger scale or multiple iterations.

It is a good question.  And we are exploring it these days. 
We are verifying the domain gap between vqkd training and beit pretraining. But it may cost some days.  
Don't worry. We will report it if we get some conclusions.

Thank you~
Zhiliang

**Comment by Basums** [comment]
Dear，
Today refer to your vq part of the code, but when training this loss has been e-7 magnitude, never_used has been 0.999, which is not normal, right?
Unlike your code, mine is not hw scale (2d) visual tokens, but I want to use 1d scale visual tokens, so I just remove norm_ema_quantizer, in forward, z = rearrange(z, 'b c h w -> b h w c'), because my input is bnc's, this should be enough to modify it, hope I can get your help.

**Comment by pengzhiliang** [comment]
> Dear, I am glad to receive your timely reply. Regarding VQ-KD, there is a hypothesis about the tokenizer obtained from knowledge distillation learning, that this teacher is able to do the classification task, but the effect is not particularly good (lacking some masks), and I hope you can tell us after your experiment to verify this conjecture. This issue can hopefully be closed after you report this conclusion, please.

Hello.
I fine-tuned the tokenzier encoder following BEiT fine-tuning recipe. For the hyper-parameters `lr`, I searched it in `{5e-4, 7e-4, 1e-3, 2e-3, 4e-3}`. The best accuracy is 83.57\% when the `lr` is `2e-3`.
Hope this can help you.

Zhiliang.

**Comment by Basums** [comment]
> > Dear, I am glad to receive your timely reply. Regarding VQ-KD, there is a hypothesis about the tokenizer obtained from knowledge distillation learning, that this teacher is able to do the classification task, but the effect is not particularly good (lacking some masks), and I hope you can tell us after your experiment to verify this conjecture. This issue can hopefully be closed after you report this conclusion, please.
> 
> Hello. I fine-tuned the tokenzier encoder following BEiT fine-tuning recipe. For the hyper-parameters `lr`, I searched it in `{5e-4, 7e-4, 1e-3, 2e-3, 4e-3}`. The best accuracy is 83.57% when the `lr` is `2e-3`. Hope this can help you.
> 
> Zhiliang.

This is very helpful to me, thank you for conducting these experiments and I hope all goes well with the follow-up.

**Comment by Basums** [comment]
Good job!

**Comment by donglixp** [comment]
The code and pre-trained models of BEiT-3 can be found at [aka.ms/beit3](https://aka.ms/beit3).
